### DISCUSSION TITLE
JSManagedValue和JscWeakRef 语义不一样

### DISCUSSION THREAD

**Discussion by breakerrorchen** [Tencent/ScriptX#80] [General]
JSManagedValue 是用来解决循环应用的，并且保证不会被GC
JscWeakRef 是不是类似c++ weak_ptr, 还是会被GC掉？

**Comment by LanderlYoung** [comment]
这些是实现细节，API上`script::Weak` 是弱引用，会被GC。

**Reply by breakerrorchen** [reply]
但是 JSManagedValue 是不会被GC的

**Reply by LanderlYoung** [reply]
I didn't get our topic yet.
If you are worried WeakRef is not implemented properly, you can do code analysis or test. And PR or ISSUE with full context is welcome.
If you want just to discuss JavaScriptCore API usage, please mark it in the title.

Thanks.
